FP_analysis_of_reviews_from_Amazon/test_nearest_review.py

```diff
@@ -0,0 +1,267 @@
+from unittest import TestCase, mock, main
+from nearest_review import (
+    ReviewTimeInfo,
+    prepare_data,
+    count_deltatime,
+    get_unix_diff_per_name,
+    get_potential_bots,
+    filter_bot_review,
+    process_data,
+    analyze_bots_comments,
+    get_nearest_reviews,
+)
+
+class TestPrepareData(TestCase):
+    def setUp(self):
+        # min set of keys in dictionaries(might be more)
+        self.data = [
+            {"reviewerID": "1", "reviewText": "a", "unixReviewTime": 200,},
+            {"reviewerID": "1", "reviewText": "b", "unixReviewTime": 300},
+            {"reviewerID": "1", "reviewText": "d", "unixReviewTime": 100},
+            {"reviewerID": "2", "reviewText": "f", "unixReviewTime": 500},
+        ]
+    def test_basic_work(self):
+        result = prepare_data(self.data)
+        self.assertIsInstance(result, dict)
+        self.assertTrue(v is ReviewTimeInfo for v in result.values())
+        self.assertCountEqual(result.keys(), ["1", "2"])
+        self.assertTrue(len(result["1"]) == 3)
+        self.assertTrue(len(result["2"]) == 1)
+        self.assertEqual(result["1"][0].time, self.data[2]["unixReviewTime"])
+        self.assertEqual(result["1"][1].time, self.data[0]["unixReviewTime"])
+        self.assertEqual(result["1"][2].time, self.data[1]["unixReviewTime"])
+        self.assertEqual(result["1"][0].text, self.data[2]["reviewText"])
+        self.assertEqual(result["1"][1].text, self.data[0]["reviewText"])
+        self.assertEqual(result["1"][2].text, self.data[1]["reviewText"])
+        self.assertTrue(
+            i.text is str or i.text is None for v in result.values() for i in v
```

Comment:
`i.text is str` - are you trying to check the type? Then please use `isinstance`. `is` is used to compare two objects by reference, not the type. Check elsewhere also

Reply:
I tried to check each element in the second position from an object from the list that it is an instance of a string or non-type object. Would it be better to split it into multiple checks for each element using assertIsInstance?

Reply:
It is fine to do either multiple `assertIsInstance` or this way:
```python
self.assertTrue(all(isinstance(i.text, str) or i.text is None for ...))
```
Your tests pass because the code is incorrect twice. `i.text` never *is* `str` (but it `isinstance` of `str`). Inside `assertTrue` you construct a generator which produces `False` for each `i`. But since generator is not empty it is considered a `True` value and the test passes. You can read more about [`__bool__`](https://docs.python.org/3/reference/datamodel.html#object.__bool__) method to understand

Reply:
please fix or ask me if you have more questions on this

Reply:
Done